c2corg_api/models/association.py

```diff
@@ -381,5 +381,6 @@ def _get_associations_to_add(new_associations, current_associations):
                    'outings', 'books', 'xreports'},
     AREA_TYPE: {'images'},
     BOOK_TYPE: {'routes', 'articles', 'images', 'waypoints'},
-    XREPORT_TYPE: {'routes', 'outings', 'articles', 'images'}
+    XREPORT_TYPE: {'routes', 'outings', 'articles', 'images', 'waypoints',
```

Comment:
I am not sure about waypoints - it is mentioned also in the issue (https://github.com/c2corg/v6_ui/pull/1108#issue-193748043). Articles can be probably also linked to Xreport from its own page.